src/XRP/common-xrpl-client.ts

```diff
@@ -0,0 +1,277 @@
+/* eslint-disable no-restricted-syntax */
+import { Signer, Utils, Wallet, XrplNetwork } from 'xpring-common-js'
+import XrpUtils from './xrp-utils'
+import GrpcNetworkClient from './grpc-xrp-network-client'
+import GrpcNetworkClientWeb from './grpc-xrp-network-client.web'
+import { XRPDropsAmount } from './Generated/web/org/xrpl/rpc/v1/amount_pb'
+import { AccountRoot } from './Generated/web/org/xrpl/rpc/v1/ledger_objects_pb'
+import {
+  Account,
+  LastLedgerSequence,
+  SigningPublicKey,
+} from './Generated/web/org/xrpl/rpc/v1/common_pb'
+import { Transaction } from './Generated/web/org/xrpl/rpc/v1/transaction_pb'
+import { AccountAddress } from './Generated/web/org/xrpl/rpc/v1/account_pb'
+import { GetFeeResponse } from './Generated/web/org/xrpl/rpc/v1/get_fee_pb'
+import TransactionStatus from './transaction-status'
+import RawTransactionStatus from './raw-transaction-status'
+import { XrpNetworkClient } from './xrp-network-client'
+import XrpError from './xrp-error'
+import { LedgerSpecifier } from './Generated/web/org/xrpl/rpc/v1/ledger_pb'
+import TransactionResult from './model/transaction-result'
+import isNode from '../Common/utils'
+
+/** A margin to pad the current ledger sequence with when submitting transactions. */
+const maxLedgerVersionOffset = 10
+
+/**
+ * CommonXrplClient is a client which interacts with the XRP Ledger.
+ */
+export default class CommonXrplClient {
```

Comment:
what other shared functionality do you expect to be here? as of now, it seems like this is kind of an `XrplTransactionClient`, but i don't know enough about xrpl to know what else there will be

is there a better name for this class? we're doing this because we expect to implement a class for sending XRP transactions and one for IssuedCurrencies, correct?

i guess what i'm wondering is...could this just be called `XrplClient`? `XrpLedgerClient`? `LedgerClient`? the class used before to send transactions is called `XrpClient`, and its new counterpart is likely `IssuedCurrenciesClient`

Reply:
So my only hesitation in renaming to `XrpTransactionClient` is that there is some common functionality such as `account_info`, which has to do with more than just transactions... and I imagine that could grow in the future.  Also, I want to use the name `XrpLedgerClient` for the top-level client that is going to compose the `IssuedCurrencyClient` and the existing `XrpClient`  (assuming we go that route).  I think keeping "common" in the name does a better job indicating that this is a bottom-level client that deals with any and all shared xrpl infrastructure

Reply:
in the end i don't rly care since all this ends up being private

"Common" strikes me as a bit of an odd prefix for some reason, but gogogo. i maybe like Core better. i don't like Base because it implies some sort of class hierarchy. if i think of something better i'll suggest it one day

Reply:
renamed to `CoreXrplClient` - I think that's great